bulldozer/signals.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bulldozer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/palantir/bulldozer/pull"
+)
+
+type Signals struct {
+	Labels            []string `yaml:"labels"`
+	CommentSubstrings []string `yaml:"comment_substrings"`
+	Comments          []string `yaml:"comments"`
+	PRBodySubstrings  []string `yaml:"pr_body_substrings"`
+	Branches          []string `yaml:"branches"`
+}
+
+func (s *Signals) Enabled() bool {
+	size := 0
+	size += len(s.Labels)
+	size += len(s.CommentSubstrings)
+	size += len(s.Comments)
+	size += len(s.PRBodySubstrings)
+	size += len(s.Branches)
+	return size > 0
+}
+
+// Matches returns true if the pull request meets one or more signals. It also
+// returns a description of the signal that was met. The mode argument appears
+// in this description and indicates the mode (whitelist, blacklist) this set
+// of signals is associated with.
+func (s *Signals) Matches(ctx context.Context, pullCtx pull.Context, mode string) (bool, string, error) {
```

Comment:
nit: `mode` as a variable name makes me think it causes different behavior. i can't come up with anything better: `modeDescription` just feels bad. the godoc should be sufficient. i just dont want to end up using the mode to drive behavior

Reply:
How about `label`?

Reply:
much like label

Reply:
Turns out `label` shadows the usage of PR labels in this function, so you're getting `tag` instead, which I think works the same.